### DISCUSSION TITLE
Caprover Instalation

### DISCUSSION THREAD

**Discussion by Bitsafeti** [chaskiq/chaskiq#783] [Q&A]
I tried to install in all ways in caprover, but I always have the same error:
In teh log: undefined method `app' for nil:NilClass

In the brownser: 
We're sorry, but something went wrong.
If you are the application owner check the logs for more information.

Looking at the error by bugsnag, it shows this:

	
  def messenger_data
    article_setting = ArticleSetting.find_by(subdomain: request.subdomain)
    @app = article_setting.app
    key = @app.encryption_key
    @sessionless = params[:sessionless]

Can anyone help me? Well, I'm almost giving up.

**Comment by michelson** [comment]
Hi, you need to add the env vars HOST and WS, so if you are going to run chaskiq on your domain  myapp.domain.com , then the environment variables should be. HOST=https://myapp.domain.com and WS wss://myapp.domain.com.
Also there are other environment variables that are described here: https://dev.chaskiq.io/production-configuration/enviroment-variables-configuration

**Reply by Bitsafeti** [reply]
Hi friend, thanks for the reply.
However, these variables are already set in the system.
I'm trying to use caprover's original url.

![url](https://user-images.githubusercontent.com/110269582/181866698-2e435cb8-cdcb-450b-9153-2a8d53750e16.png)
![variaveis](https://user-images.githubusercontent.com/110269582/181866699-6ac5eee1-8c33-4794-96e9-a993bc3f19ec.png)

**Reply by michelson** [reply]
try to set a domain with one level, like **app.domain.com**

**Reply by Bitsafeti** [reply]
some error my friend.
![mesmo erro](https://user-images.githubusercontent.com/110269582/181866826-d0af51ab-a097-4fe8-acc1-0595fe86928e.png)
![Novo dominio](https://user-images.githubusercontent.com/110269582/181866827-431d4302-81fb-4f8c-8d67-cc3ed15b5ca7.png)
![variaveis2](https://user-images.githubusercontent.com/110269582/181866828-1a9f160c-efdf-439c-9332-08f8bff95ae1.png)

**Reply by Bitsafeti** [reply]
More Logs
![image](https://user-images.githubusercontent.com/110269582/181867022-40255572-0b10-4e4d-8960-5b0119f85ac5.png)

**Reply by michelson** [reply]
Uhm.. maybe there is a problem with domains with multiple TLD names, like `com.br`, and it could be happening that is recognizing the subdomain wrong. Do you have another domain to test with?

**Reply by Bitsafeti** [reply]
@michelson 
Friend tested using a .com domain and it worked.
But I'm from Brazil, wouldn't there be a way to fix this bug?

![image](https://user-images.githubusercontent.com/110269582/181868379-e5b7d156-edcc-44df-9b35-6815d354a96f.png)

**Reply by michelson** [reply]
absolutely, we are going to fix this ASAP. I did a small test and saw that in your case there will be two subdomains detected, like: ["app", "chat"], that's basically the issue here.

**Reply by Bitsafeti** [reply]
Understood friend, I'm waiting for the correction.
However, I have a call with you on Monday 01/08/2022.
We talk over there.
Big hug.

**Reply by michelson** [reply]
great, see you next week. The fix is on its way #784
